py-rattler-build/src/lib.rs

```diff
@@ -37,6 +45,174 @@ where
     })
 }
 
+/// Python wrapper for SelectorConfig
+#[pyclass]
+#[derive(Clone)]
+pub struct PySelectorConfig {
+    inner: SelectorConfig,
+}
+
+#[pymethods]
+impl PySelectorConfig {
+    #[new]
+    #[pyo3(signature = (target_platform=None, host_platform=None, build_platform=None, variant=None, experimental=None, allow_undefined=None, recipe_path=None))]
+    fn new(
+        target_platform: Option<String>,
+        host_platform: Option<String>,
+        build_platform: Option<String>,
+        variant: Option<HashMap<String, Py<PyAny>>>,
+        experimental: Option<bool>,
+        allow_undefined: Option<bool>,
+        recipe_path: Option<PathBuf>,
+    ) -> PyResult<Self> {
+        let target_platform = target_platform
+            .map(|p| Platform::from_str(&p))
+            .transpose()
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?
+            .unwrap_or_else(Platform::current);
+
+        let host_platform = host_platform
+            .map(|p| Platform::from_str(&p))
+            .transpose()
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?
+            .unwrap_or_else(Platform::current);
+
+        let build_platform = build_platform
+            .map(|p| Platform::from_str(&p))
+            .transpose()
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?
+            .unwrap_or_else(Platform::current);
+
+        // Convert variant from Python dict to BTreeMap<NormalizedKey, Variable>
+        let variant_map = if let Some(variant_dict) = variant {
+            Python::attach(|py| {
```

Comment:
you can receive py: Python as an argument in any `#[pymethods]` so you don't need anymore to `attach` or request for gil token